common/instance_config/puppet.yaml

```diff
@@ -66,6 +66,12 @@ runcmd:
   - grub2-mkconfig -o /boot/grub2/grub.cfg
 # Remove all ifcfg configuration files that have no corresponding network interface in ip link show.
   - for i in /etc/sysconfig/network-scripts/ifcfg-*; do if ! ip link show | grep -q "$${i##*-}:"; then rm -f $i; fi; done
+# AWS EFA driver installation
+%{ if contains(tags, "efa") }
```

Comment:
I did another test with a mix of EFA and non-EFA nodes. If the EFA drivers are not installed, a simple bandwidth test gives
```
# OSU MPI Bandwidth Test v5.6.3
# Size      Bandwidth (MB/s)
[node1.int.aws2.learnhpc.eu:08156] mtl_ofi_endpoint.h:47: *** The Open MPI OFI MTL is aborting the MPI job (via exit(3)).
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[21688,1],0]
  Exit code:    1
--------------------------------------------------------------------------
```
but if we just install the EFA drivers on the non-EFA nodes we get a much clearer error:
```
[node1.int.aws2.learnhpc.eu:08559] pml_ucx.c:273  Error: Failed to create UCP worker
# OSU MPI Latency Test v5.6.3
# Size          Latency (us)
[node1.int.aws2.learnhpc.eu:08559] mtl_ofi_endpoint.h:47: *** The Open MPI OFI MTL is aborting the MPI job (via exit(3)).
[tnode1.int.aws2.learnhpc.eu:02146] [[27148,1],1] selected pml ucx, but peer [[27148,1],0] on node1 selected pml cm
[tnode1.int.aws2.learnhpc.eu:02146] pml_ucx.c:452  Error: Failed to resolve UCX endpoint for rank 0
```
I think it makes sense then to install the EFA drivers if _anyone_ is using EFA.

Reply:
Do you think it still makes sense to install it everywhere now that we found the problem was the memlock security limits?

Reply:
No, I think either updates the limits on login nodes or adjust the SLURM configuration is all that's required. Will test both tomorrow.